reconcilers/reconcilers.go

```diff
@@ -396,11 +396,11 @@ func (r *SyncReconciler) validate(ctx context.Context) error {
 	//     func(ctx context.Context, parent client.Object) error
 	//     func(ctx context.Context, parent client.Object) (ctrl.Result, error)
 	if r.Sync == nil {
-		return fmt.Errorf("SyncReconciler must implement Sync")
+		return fmt.Errorf("SyncReconciler %q must implement Sync", r.Name)
```

Comment:
Unlike ChildReconciler, SyncReconciler doesn't have a default Name currently. This will create default errors like `SyncReconciler "" must implement Sync`.

Do you think this is ok? Should we define a default name?

Reply:
All reconcilers have a name as of #231